pvanalytics/features/snow.py

```diff
@@ -0,0 +1,266 @@
+             of the inverter's MPPT algorithm OR
+          3. measured voltage exceeds the upper bound of the inverter's MPPT
+             algorithm.
+
+      Mode -1 is added in this function to cover a case that was not addressed
+      in [1]_.
+
+    * Mode 0: Indicates periods with enough opaque snow that the system is not
+      producing power. Specifically, Mode 0 is when the measured voltage is
+      below the inverter's turn-on voltage but the voltage modeled using
+      measured irradiance is above the inverter's turn-on voltage.
+    * Mode 1: Indicates periods when the system has non-uniform snow and
+      both operating voltage and current are decreased. Operating voltage is
+      reduced when bypass diodes activate and current is decreased due to
+      decreased irradiance.
+    * Mode 2: Indicates periods when the operating voltage is reduced but
+      current is consistent with snow-free conditions.
+    * Mode 3: Indicates periods when the operating voltage is consistent with
+      snow-free conditionss but current is reduced.
+    * Mode 4: Voltage and current are consistent with snow-free conditions.
+
+    Parameters
+    ----------
+    transmission : array-like
+        Fraction of plane-of-array irradiance that can reach the array's cells
+        through the snow cover. [dimensionless]
+    measured_voltage : array-like
+        Measured DC voltage. [V]
+    modeled_voltage_with_calculated_transmission : array-like
+        DC voltage modeled using measured plane-of-array irradiance,
+        back-of-module temperature, and calculated transmission. [V]
+    modeled_voltage_with_ideal_transmission : array-like
+        DC voltage modeled using measured plane-of-array irradiance and
+        back-of-module temperature. Assumes transmission equals 1. [V]
+    min_dcv : float
+        The lower voltage bound on the inverter's maximum power point
+        tracking (MPPT) algorithm. [V]
+    max_dcv : float
+        The upper voltage bound on the inverter's maximum power point
+        tracking (MPPT) algorithm. [V]
+    threshold_vratio : float
+        The lower bound for vratio that is representative of snow-free
+        conditions. Determined empirically. Depends on system configuration and
+        site conditions. [unitless]
+    threshold_transmission : float
+        The lower bound on transmission that is found under snow-free
+        conditions, determined empirically. [unitless]
+
+    Returns
+    -------
+    mode : int or None
+        ``mode`` is ``None`` when any of the inputs used to determine ``mode``
+        is ``nan``.
+
+    References
+    ----------
+    .. [1] E. C. Cooper, J. L. Braid and L. M. Burnham, "Identifying the
+       Electrical Signature of Snow in Photovoltaic Inverter Data," 2023 IEEE
+       50th Photovoltaic Specialists Conference (PVSC), San Juan, PR, USA,
+       2023, pp. 1-5, :doi:`10.1109/PVSC48320.2023.10360065`.
+    """
+    umin_meas = measured_voltage >= min_dcv
```

Comment:
A comment at the top of this function's code to explain the "u" prefix in the variable names might be helpful for new readers

Reply:
The "u" is an old habit of mine for naming variables that are arrays of logicals. Unnecessary here.